README.md

```diff
@@ -39,6 +39,86 @@ If you have previous version of PSScriptAnalyzer installed on your machine, you
 
 To confirm installation: run ```Get-ScriptAnalyzerRule``` in the PowerShell console to obtain the built-in rules
 
+Suppressing Rules
+=================
+
+You can suppress a rule by decorating a script/function or script/function parameter with .NET's [SuppressMessageAttribute](https://msdn.microsoft.com/en-us/library/system.diagnostics.codeanalysis.suppressmessageattribute.aspx).  `SuppressMessageAttribute`'s constructor takes two parameters: a category and a check ID. Set the `categoryID` parameter to the name of the rule you want to suppress (you may omit the `checkID` parameter):
+
+    function SuppressMe()
+    {
+        [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSProvideCommentHelp")]
+        param()
+
+        Write-Verbose -Message "I'm making a difference!"
+        
+    }
+    
+To suppress a message on a specific parameter, set the `SuppressMessageAttribute`'s `CheckId` parameter to the name of the parameter:
+
+    function SuppressTwoVariables()
+    {
+        [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSProvideDefaultParameterValue", "b")]
+        [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSProvideDefaultParameterValue", "a")]
+        param([string]$a, [int]$b)
+        {
+        }
+    }
+    
+To suppress a rule for an entire function/script, decorate the `param` block of the script/function and set the `SuppressMessageAttribute's` `Scope` property to `Function`:
+
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSProvideCommentHelp", "", Scope="Function")]
+    param(
+    )
+    
+
+You can also suppress a rule for an entire class using `Class` as the value of the `Scope` property:
```

Comment:
So the scope reflects the kind of objects in the scope where the suppress
attribute is? So a scope of "function" would apply to all functions in the
scope of the attribute?

On Mon, Jul 13, 2015 at 3:40 PM, Quoc Truong <notifications@github.com>
wrote:

> In README.md
> <https://github.com/PowerShell/PSScriptAnalyzer/pull/274#discussion_r34520293>
> :
>
> > +    {
> > +        [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSProvideDefaultParameterValue", "b")]
> > +        [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSProvideDefaultParameterValue", "a")]
> > +        param([string]$a, [int]$b)
> > +        {
> > +        }
> > +    }
> > +
> > +To suppress a rule for an entire function/script, decorate the `param` block of the script/function and set the `SuppressMessageAttribute's` `Scope` property to `Function`:
> > +
> > +    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSProvideCommentHelp", "", Scope="Function")]
> > +    param(
> > +    )
> > +
> > +
> > +You can also suppress a rule for an entire class using `Class` as the value of the `Scope` property:
>
> You can actually do this by simply decorating the class with the
> SuppressMessageAttribute (the scope is unnecessary in this case). A case
> where the use of scope is more meaningful is if you want to suppress a rule
> for all the classes within a script, then you can decorate the script with
> a SuppressMessageAttribute with the Scope as Class.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PowerShell/PSScriptAnalyzer/pull/274/files#r34520293>.
>